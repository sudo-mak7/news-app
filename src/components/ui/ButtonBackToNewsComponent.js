import React from 'react'
import { Button, Icon } from 'semantic-ui-react'
import { useNavigate  } from 'react-router-dom'

const ButtonBackToNewsComponent = () => {
  const navigate = useNavigate()

  return (
    <Button
      icon
      size='big'
      color='blue'
      style={{
        zIndex: '999',
        position: 'absolute',
        top: '0.25em',
        left: '0.25em'
      }}
      onClick={ () => { navigate(-1) } }
    >
      <Icon name='arrow left'></Icon>
    </Button>
  )
}

export default ButtonBackToNewsComponent