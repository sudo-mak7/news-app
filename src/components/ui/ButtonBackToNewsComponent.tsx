import * as React from 'react'
import { Button, Icon } from 'semantic-ui-react'
import { useNavigate  } from 'react-router-dom'

const ButtonBackToNewsComponent = (): JSX.Element => {
  const navigate = useNavigate()

  return (
    <Button
      icon
      size='big'
      color='blue'
      style={{
        zIndex: '999',
        position: 'fixed',
        top: '0.25em',
        left: '0.25em'
      }}
      onClick={ (): void => { navigate(-1) } }
    >
      <Icon name='arrow left'></Icon>
    </Button>
  )
}

export default ButtonBackToNewsComponent