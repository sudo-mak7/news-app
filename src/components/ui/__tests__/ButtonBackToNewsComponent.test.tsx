import * as React from 'react'
import * as routerHooks from 'react-router-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, screen, fireEvent } from '@testing-library/react'
import ButtonBackToNewsComponent from '@components/ui/ButtonBackToNewsComponent'

const buttonBackToNewsComponent = (): JSX.Element =>
  <BrowserRouter>
    <ButtonBackToNewsComponent />
  </BrowserRouter>

jest.mock(
  'react-router-dom',
  () => ({
    __esModule: true, ...jest.requireActual('react-router-dom')
  })
)

const mockedNavigate = jest.spyOn(routerHooks, 'useNavigate')

describe('ButtonBackToNewsComponent test', () => {
  test('renders ButtonBackToNewsComponent', () => {
    render(buttonBackToNewsComponent())
    expect(buttonBackToNewsComponent()).toMatchSnapshot()

    const buttonInButtonBackToNewsComponent = screen.getByRole('button')
    expect(buttonInButtonBackToNewsComponent).toHaveClass('ui blue big icon button')
  })

  test('navigate(-1) should called once by click with -1', () => {
    const navigate = jest.fn()
    mockedNavigate.mockImplementation(() => navigate)

    render(buttonBackToNewsComponent())
    const buttonInButtonBackToNewsComponent = screen.getByRole('button')

    fireEvent.click(buttonInButtonBackToNewsComponent)
    expect(navigate).toHaveBeenCalledWith(-1)
  })

  afterAll(() => {
    jest.clearAllMocks()
  })
})