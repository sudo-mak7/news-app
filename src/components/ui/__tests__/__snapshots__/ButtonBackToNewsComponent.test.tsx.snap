// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBackToNewsComponent test renders ButtonBackToNewsComponent 1`] = `
<BrowserRouter>
  <ButtonBackToNewsComponent />
</BrowserRouter>
`;
